icepyx/core/query.py

```diff
@@ -646,6 +646,27 @@ def subsetparams(self, **kwargs):
         if self._subsetparams == None and not kwargs:
             return {}
         else:
+            # If the user has supplied a subset list of variables, append the
+            # icepyx required variables to the Coverage dict
+            if 'Coverage' in kwargs.keys():
+                var_list = [
+                    "sc_orient",
+                    "sc_orient_time",
+                    "atlas_sdp_gps_epoch",
+                    "cycle_number",
+                    "rgt",
+                    "data_start_utc",
+                    "data_end_utc",
+                    "granule_start_utc",
+                    "granule_end_utc",
+                    "start_delta_time",
+                    "end_delta_time",
+                ]
+                self.order_vars.append(defaults=False, var_list=var_list)
+                for var, path in self.order_vars.wanted.items():
+                    if var not in kwargs['Coverage'].keys():
+                        kwargs['Coverage'][var] = path
```

Comment:
I think this is redundant with [`self.order_vars.append()`](https://github.com/icesat2py/icepyx/blob/a184d9cd7bece9da303b3998d841bdcb641a5233/icepyx/core/variables.py#L500)

Reply:
Good catch. I just removed the `append()` call. Just to make sure, though, I _think_ that the `append()` call was adding the required variables to the wanted variables list, meaning that it wouldn't impact the API call to NSIDC. The loop, on the other hand, was adding the required variables to the coverage variables, meaning it is the one that has an impact on the actual data the user will get. This is why I removed the `append()` instead of the loop. Does that sound correct?

Reply:
Glad you spelled this out - I *was* mistakenly thinking that adding it to the wanted variables list would then get it to the API call. However, in this current iteration I'm not sure we'd be properly adding the `var_list` items because we never append them to `self.order_vars.wanted`. The two approaches coming to mind are (1) `self.order_vars.append(var_list)` (which will do a similar loop to update keys/values) and `kwargs['Coverage'] = self.order_vars.wanted` or (2) create a new vars object, append the `var_list` and run through this loop (though I'm not sure if we'd run into issues if the key is in there, but the value has changed). The only potential downside of the former is that it's changing the user's curated list, but I'm guessing that shouldn't be too big of a problem (plus it means they get an accurate list of variables ordered and it's a lot easier to implement).

Reply:
Thanks for thinking this through with me. I think I have something working, which I tested by requesting a subset file and checked to make sure the un-requested but required variables also got added. Look good?